matsim/src/main/java/ch/sbb/matsim/routing/pt/raptor/DefaultRaptorStopFinder.java

```diff
@@ -110,105 +102,125 @@ public DefaultRaptorStopFinder(Population population, RaptorIntermodalAccessEgre
 		double x = facility.getCoord().getX();
 		double y = facility.getCoord().getY();
 		List<InitialStop> initialStops = new ArrayList<>();
-		for (IntermodalAccessEgressParameterSet paramset : srrCfg.getIntermodalAccessEgressParameterSets()) {
-			String mode = paramset.getMode();
-			String linkIdAttribute = paramset.getLinkIdAttribute();
-			String personFilterAttribute = paramset.getPersonFilterAttribute();
-			String personFilterValue = paramset.getPersonFilterValue();
-			String stopFilterAttribute = paramset.getStopFilterAttribute();
-			String stopFilterValue = paramset.getStopFilterValue();
+        switch (srrCfg.getIntermodalAccessEgressModeSelection()) {
+            case CalcLeastCostModePerStop:
+                for (IntermodalAccessEgressParameterSet parameterSet : srrCfg.getIntermodalAccessEgressParameterSets()) {
+                    addInitialStopsForParamSet(facility, person, departureTime, direction, parameters, data, x, y, initialStops, parameterSet);
+                }
+                break;
+            case RandomSelectOneModePerRoutingRequestAndDirection:
+                int counter = 0;
+                do {
+                    int rndSelector = (int) (MatsimRandom.getRandom().nextDouble() * srrCfg.getIntermodalAccessEgressParameterSets().size());
```

Comment:
This will be non-deterministic, as all the threads access the same random number generator.